### PR TITLE
[Merged by Bors] - load spirv using correct API

### PR DESCRIPTION
**Pull request by molikto**
# Objective

currently spirv source loading parse then desterilize it to make wgpu happy.

## Solution

Use the right API to get spirv

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

File: crates/bevy_render/src/render_resource/shader.rs [hunk 1]
```diff
@@ -9,6 +9,7 @@ use std::{
     borrow::Cow, collections::HashSet, marker::Copy, ops::Deref, path::PathBuf, str::FromStr,
 };
 use thiserror::Error;
+use wgpu::util::make_spirv;
 use wgpu::{ShaderModuleDescriptor, ShaderSource};
```
Comment (mockersf) [inline crates/bevy_render/src/render_resource/shader.rs:13]: could you merge those two?

File: crates/bevy_render/src/render_resource/shader.rs [hunk 2]
```diff
@@ -172,12 +172,7 @@ impl ProcessedShader {
                     let wgsl = reflection.get_wgsl()?;
                     ShaderSource::Wgsl(wgsl.into())
                 }
-                ProcessedShader::SpirV(_) => {
-                    // TODO: we can probably just transmute the u8 array to u32?
-                    let reflection = self.reflect()?;
-                    let spirv = reflection.get_spirv()?;
-                    ShaderSource::SpirV(Cow::Owned(spirv))
-                }
+                ProcessedShader::SpirV(source) => make_spirv(source),
```
Comment (bjorn3) [inline crates/bevy_render/src/render_resource/shader.rs:175]: Does this function support spirv, wgsl and glsl inputs?

Comment (mockersf) [reply crates/bevy_render/src/render_resource/shader.rs:175]: probably not https://docs.rs/wgpu/latest/wgpu/util/fn.make_spirv.html

**Review by mockersf — APPROVED** [review]

File: crates/bevy_render/src/render_resource/shader.rs (hunk 2 above)
Comment (bjorn3) [reply crates/bevy_render/src/render_resource/shader.rs:175]: Indeed. According to the source it only allows spirv.

**Review by alice-i-cecile — APPROVED** [review]

**Pull request closed without merging**